dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java

```diff
@@ -1870,4 +1856,42 @@ public boolean isInheritingPermissions(Permissionable permissionable) throws Dot
 		return permissionFactory.isInheritingPermissions(permissionable);
 	}
 
+	/**
+	 * Retrieves a filtered list of roles by removing front-end roles when unnecessary.
+	 **/
+	private Set<Role> filterUserRoles(User user, boolean respectFrontendRoles) throws DotDataException {
+		final Role anonymousRoleRole = APILocator.getRoleAPI().loadCMSAnonymousRole();
+		final Role frontEndUserRole = APILocator.getRoleAPI().loadLoggedinSiteRole();
+
+		final Set<Role> roles = new HashSet<>(Try
+				.of(() -> APILocator.getRoleAPI().loadRolesForUser(user.getUserId()))
+				.getOrElse(List.of()));
+
+		if (user.isAnonymousUser()) {

```

Comment:
I think I would keep this comment around for future references
// remove front end user access for anon user (e.g, /intranet)
// Note to selves: it was a mistake to add this role to the Anon user in 5.2.0

Reply:
Done, thanks for the comment!